### ISSUE TITLE
I'm missing something: --inject-env isn't injecting

### ISSUE THREAD

**Issue by erik-at-techsanity** [pantsbuild/pex#2224]
When I build a simple pexfile using --inject-env:

`(venv) erik@Eriks-MBP /t/unpack> pex --inject-env FOO=BLARG --output-file /tmp/test.pex`

It doesn't seem to be present when I start up the repl:

```
(venv) erik@Eriks-MBP /t/unpack> /tmp/test.pex
Python 3.11.4 (v3.11.4:d2340ef257, Jun  6 2023, 19:15:51) [Clang 13.0.0 (clang-1300.0.29.30)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> import os
>>> os.environ.get("FOO")
>>> ^D
now exiting InteractiveConsole...
(venv) erik@Eriks-MBP /t/unpack>
```

But it's present in the PEX-INFO:

```
(venv) erik@Eriks-MBP /t/unpack> cat PEX-INFO | jq .inject_env
{
  "FOO": "BLARG"
}
```

What am I doing wrong? Thanks for any suggestions.

**Comment by jsirois** [comment]
As you say:
```
$ pex --inject-env FOO=BLARG -- -c 'import os; print(os.environ["FOO"])'
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/tmp/tmp2zj_60se/__main__.py", line 108, in <module>
    bootstrap_pex(__entry_point__, execute=__execute__, venv_dir=__venv_dir__)
  File "/tmp/tmp2zj_60se/.bootstrap/pex/pex_bootstrapper.py", line 627, in bootstrap_pex
    pex.PEX(entry_point).execute()
  File "/tmp/tmp2zj_60se/.bootstrap/pex/pex.py", line 560, in execute
    sys.exit(self._wrap_coverage(self._wrap_profiling, self._execute))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/tmp2zj_60se/.bootstrap/pex/pex.py", line 467, in _wrap_coverage
    return runner(*args)
           ^^^^^^^^^^^^^
  File "/tmp/tmp2zj_60se/.bootstrap/pex/pex.py", line 498, in _wrap_profiling
    return runner(*args)
           ^^^^^^^^^^^^^
  File "/tmp/tmp2zj_60se/.bootstrap/pex/pex.py", line 581, in _execute
    return self.execute_interpreter()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/tmp2zj_60se/.bootstrap/pex/pex.py", line 634, in execute_interpreter
    return self.execute_content("-c <cmd>", content, argv0="-c")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/tmp2zj_60se/.bootstrap/pex/pex.py", line 745, in execute_content
    return cls.execute_ast(name, program, argv0=argv0)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/tmp2zj_60se/.bootstrap/pex/pex.py", line 763, in execute_ast
    exec_function(program, globals_map)
  File "/tmp/tmp2zj_60se/.bootstrap/pex/compatibility.py", line 109, in exec_function
    exec (ast, globals_map, locals_map)
  File "-c <cmd>", line 1, in <module>
  File "<frozen os>", line 685, in __getitem__
KeyError: 'FOO'
```

But:
```
$ echo 'import os; print(os.environ["FOO"])' > exe.py
$ pex --inject-env FOO=BLARG --exe exe.py
BLARG
```

The issue is injection is done down on lines 596-598 _after_ interpreter mode is dispatched:
https://github.com/pantsbuild/pex/blob/10f3eb8250b686b3ad0bae898254010046c6b1a8/pex/pex.py#L562-L605

Looking back at the original issue (#987) and PR where this landed (#1948) I cannnot see why I did this. Perhaps an easy fix. Are you game @erik-at-techsanity?

**Comment by erik-at-techsanity** [comment]
absolutely - and thanks for the digging you did - I'll get this moving.

**Comment by jsirois** [comment]
@erik-at-techsanity I'll be AFK through August 28th but can take a look at any progress you've made then.

**Comment by jsirois** [comment]
@erik-at-techsanity - just checking back in to see if you still want to work on this or if I should release it back to the pool.

**Comment by erik-at-techsanity** [comment]
Hi John – Apologies! I’ll absolutely take it on – hoping to get on it tomorrow.


From: John Sirois ***@***.***>
Date: Thursday, September 21, 2023 at 3:48 PM
To: pantsbuild/pex ***@***.***>
Cc: Erik Williamson ***@***.***>, Mention ***@***.***>
Subject: Re: [pantsbuild/pex] I'm missing something: --inject-env isn't injecting (Issue #2224)

@erik-at-techsanity<https://github.com/erik-at-techsanity> - just checking back in to see if you still want to work on this or if I should release it back to the pool.

**Comment by erikwilliamson** [comment]
ok - I've created https://github.com/pantsbuild/pex/pull/2260 - totally new to contributing (to anything...) so please let me know if I've missed any steps, etc. Looks like you'll need to approve the checks to run, right?

**Comment by jsirois** [comment]
Thanks @erikwilliamson - I added instructions on the PR for running tests. I'll let CI fly once you've added some.

**Comment by jsirois** [comment]
Closed bby #2260
